src/pages/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from 'react';
+import GenreSelect from '../../components/GenreSelect/GenreSelect';
+import Header from '../../components/Header/Header';
+import MovieList from '../../components/MovieList/MovieList';
+import SortControl from '../../components/SortControl/SortControl';
+import SortOrderControl from '../../components/SortOrder/SortOrderControl';
+import { useFetch } from '../../hooks/useFetch';
+import { genresMock } from '../../mocks/Genre';
+import { Movie, MoviesRequest, SortBy, SortOrder } from '../../models/Movie';
+import MovieService from '../../services/MovieService';
+import { getPageCount } from '../../utils/PageUtils';
+import './MovieListPage.scss';
+
+function MovieListPage() {
+  const movieService = new MovieService();
+  const genres = ['All', ...genresMock];
+  const limit = 6;
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [totalAmount, setTotalAmount] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
+  const [page, setPage] = useState(1);
+  const [sortBy, setSortBy] = useState<SortBy>('release_date');
+  const [sortOrder, setSortOrder] = useState<SortOrder>('desc');
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedMovie, setSelectedMovie] = useState<Movie | null>(null);
+  const [selectedGenre, setSelectedGenre] = useState<string>(genres[0]);
+
+  const prepareRequestParams = (): MoviesRequest => {
+    return {
+      search: searchQuery,
+      searchBy: 'title',
+      sortBy: sortBy,
+      sortOrder: sortOrder,
+      limit: limit.toString(),
+      offset: (movies.length).toString(),
+      filter: selectedGenre !== 'All' ? selectedGenre : ''
+    }
+  }
+
+  const [fetchMovies, moviesLoading, moviesError] = useFetch(async () => {
+    const response = await movieService.getMovies(prepareRequestParams());
+    setMovies([...movies, ...response.data]);
+    setTotalAmount(response.totalAmount);
+    setTotalPages(getPageCount(response.totalAmount, limit));
+  })
+
+  useEffect(() => {
+    (async () => {
+      await fetchMovies();
+    })();
```

Comment:
Since the result is not used:
```suggestion
    void fetchMovies();
```

Reply:
Looks much better in this way!